jobbergate-agent/jobbergate_agent/internals/update.py

```diff
@@ -0,0 +1,71 @@
+import subprocess
+import sys
+
+from loguru import logger
+from pkg_resources import get_distribution
+
+from jobbergate_agent.clients.cluster_api import backend_client as jobbergate_api_client
+from jobbergate_agent.utils.scheduler import schedule_tasks, scheduler
+
+
+package_name = "jobbergate_agent"
+
+
+async def _fetch_upstream_version_info() -> str:
+    logger.debug("Fetching the upstream version info from jobbergate API")
+    response = await jobbergate_api_client.get("/jobbergate/openapi.json")
+    response.raise_for_status()
+
+    data = response.json()
+    upstream_version: str = data["info"]["version"]
+    return upstream_version
+
+
+def _compare_versions(current_version: str, upstream_version: str) -> bool:
```

Comment:
`need_update` could be a more representative name for this function

Reply:
good feedback, thank you!

Reply:
see [6d2a394](https://github.com/omnivector-solutions/jobbergate/pull/518/commits/6d2a394811dbdbbb57703f3d72115e8791ce1899)